### ISSUE TITLE
asyncFork returns invalid iterator in 7.1 

### ISSUE THREAD

**Issue by ZhangYiJiang** [iter-tools/iter-tools#463]
Version 7.1 has a new implementation of `asyncFork` that appears to return invalid iterators. This is a simplified version of our code that experiences this issue 

``` ts 
import { asyncFork, asyncCycle, range, asyncFind } from "iter-tools-es";

(async () => {
  const iterator = asyncCycle(range(100));
  const [a] = asyncFork(iterator);
  const result = await asyncFind((x) => x > 5, a);
  console.log(result);
})();
``` 

- Version 7.0 - https://codesandbox.io/s/priceless-ramanujan-liehbh?file=/src/index.js no error correctly returns 6
- Version 7.1 - https://codesandbox.io/s/optimistic-drake-jryx63?file=/src/index.js crashes with "TypeError iterator.return() returned a non-object value"

**Comment by conartist6** [comment]
Huh. It looks like the error is actually about `return()` not returning an iterator step object. I wonder why my tests didn't catch it?

**Comment by conartist6** [comment]
I think it must be because it's implementation specific what the for loop does when return is not a step. Anyway, the fix is easy.

**Comment by conartist6** [comment]
Fixed in `iter-tools@7.2.2`

**Comment by abrenneke** [comment]
Thanks for the quick fix!
